relay-server/src/actors/upstream.rs

```diff
@@ -881,69 +872,108 @@ impl Entry {
 /// is synchronous and managed by the [`UpstreamBroker`].
 #[derive(Debug)]
 struct UpstreamQueue {
+    /// High priority queue.
     high: VecDeque<Entry>,
+    /// Low priority queue.
     low: VecDeque<Entry>,
+    /// High priority retry queue.
+    retry_high: VecDeque<Entry>,
+    /// Low priority retry queue.
+    retry_low: VecDeque<Entry>,
+    /// Retries should not be dequeued before this instant.
+    ///
+    /// This retry increments by the constant `retry_after` instead of backoff,
+    /// since it only kicks in for a short period of time before Relay gets into
+    /// network outage mode (see [`IsNetworkOutage`]).
+    next_retry: Instant,
+    /// Time to wait before retrying another request from the retry queue.
+    retry_after: Duration,
```

Comment:
To make it clearer I would have it `interval`

```suggestion
    retry_interval: Duration,
```